crates/iota-data-ingestion/src/workers/blob.rs

```diff
@@ -2,31 +2,132 @@
+}
+
+fn deserialize_chunk<'de, D>(deserializer: D) -> Result<u64, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let checkpoint_chunk_size = u64::deserialize(deserializer)? * 1024 * 1024;
+    if checkpoint_chunk_size < MIN_CHUNK_SIZE_MB {
+        return Err(serde::de::Error::custom("Chunk size must be at least 5 MB"));
+    }
+    Ok(checkpoint_chunk_size)
 }
 
 pub struct BlobWorker {
     remote_store: Box<dyn ObjectStore>,
+    checkpoint_chunk_size_mb: u64,
 }
 
 impl BlobWorker {
     pub fn new(config: BlobTaskConfig) -> Self {
         Self {
-            remote_store: create_remote_store_client(config.url, config.remote_store_options, 10)
-                .expect("failed to create remote store client"),
+            checkpoint_chunk_size_mb: config.checkpoint_chunk_size_mb,
+            remote_store: create_remote_store_client_with_ops(
+                config.url,
+                config.remote_store_options,
+                config.request_timeout_secs,
+                RetryConfig {
+                    max_retries: 10,
+                    retry_timeout: Duration::from_secs(config.request_timeout_secs + 1),
+                    backoff: BackoffConfig {
+                        init_backoff: Duration::from_secs(1),
+                        ..Default::default()
+                    },
+                },
+            )
+            .expect("failed to create remote store client"),
+        }
+    }
+
+    /// Uploads a Checkpoint blob to the Remote Store.
+    ///
+    /// If the blob size exceeds the configured `CHUNK_SIZE`,
+    /// it uploads the blob in parts using multipart upload.
+    /// Otherwise, it uploads the blob directly.
+    async fn upload_blob(&self, bytes: Vec<u8>, chk_seq_num: u64, location: Path) -> Result<()> {
+        if bytes.len() > self.checkpoint_chunk_size_mb as usize {
+            return self
+                .upload_blob_multipart(bytes, chk_seq_num, location)
+                .await;
         }
+
+        self.remote_store
+            .put(&location, Bytes::from(bytes).into())
+            .await?;
+
+        Ok(())
+    }
+
+    /// Uploads a large Checkpoint blob to the Remote Store using multipart
+    /// upload.
+    ///
+    /// This function divides the input `bytes` into chunks of size `CHUNK_SIZE`
+    /// and uploads each chunk individually.
+    /// Finally, it completes the multipart upload by assembling all the
+    /// uploaded parts.
+    async fn upload_blob_multipart(
+        &self,
+        bytes: Vec<u8>,
+        chk_seq_num: u64,
+        location: Path,
+    ) -> Result<()> {
+        let mut multipart = self.remote_store.put_multipart(&location).await?;
+        let chunks = bytes.chunks(self.checkpoint_chunk_size_mb as usize);
+        let total_chunks = chunks.len();
+
+        let mut parts_ordered_futures = chunks
+            .into_iter()
+            .map(|chunk| multipart.put_part(Bytes::copy_from_slice(chunk).into()))
+            .collect::<FuturesUnordered<_>>();
```

Comment:
Nit: You could use the `enumerate` adaptor and get rid the `uploaded_part` counter.

Reply:
[fixed](https://github.com/iotaledger/iota/pull/5026/commits/2c8ce076368585f9cc265f52d40125b79804f7d0#diff-2d923450ac72bc499c7474995e7f7811e449b1baa460bd5644c03b58f7d3d60dR107-R112)